### PR TITLE
Added board,player_control, and tests to Piece

### PR DESCRIPTION
**Pull request by ianm24**

### CODE REVIEW

**Comment by ianm24** [comment]
Closes #10

File: tests/unit/test_piece.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+def test_perform_invalid_action():
+    '''Ensures proper function of perform_action with valid input'''
+    p = Piece('t', 1, Board.empty())
+    assert p.perform_action("test", []) is False
+
+
+def test_perform_invalid_move_args_action():
+    '''Ensures proper function of perform_action with invalid move args'''
+    p = Piece('t', 1, Board.empty())
+    assert p.perform_action("move", ["t", "1"]) is False and p.perform_action(
```
Comment (jesp1999) [inline tests/unit/test_piece.py:167]: Might be worth breaking these up into multiple statements for readability:
```
assert p.perform_action("move", ["t", "1"]) is False
assert p.perform_action("move", ["1", "t") is False
assert p.current_coords == (0,0)
```

File: tests/unit/test_piece.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+#     with pytest.raises(NonCurrentPiecePlayerNumException):
+#         Piece('t', 1, Board(1, 1, [""],
+#                             current_players=[0, 1, 2]), player_controller=3)
+
+
+def test_player_controller():
+    """Ensures proper function when a piece has valid player controller"""
+#         Piece('t', 1, Board(1, 1, [""],
+#                             current_players=[0, 1, 2]), player_controller=3)
+    p = Piece('t', 1, Board.empty(), player_controller=2)
```
Comment (jesp1999) [inline tests/unit/test_piece.py:114]: Btw you can do `@pytest.mark.skip(reason=REASON)` to skip running a test or `@pytest.mark.xfail(reason=REASON)` to run the test but expect it to fail (if it succeeds it'll be marked like a failure, if it fails it'll be effectively ignored)

**Review by jesp1999 — CHANGES_REQUESTED** [review]
A few change requests

**Review by jesp1999 — APPROVED** [review]

**Pull request merged**